api/turing/service/router_deployment_service.go

```diff
@@ -546,12 +566,22 @@ func deployKnServices(
 ) error {
 	// Define deploy function
 	deployFunc := func(ctx context.Context,
-		svc *cluster.KnativeService,
+		resource any,
 		wg *sync.WaitGroup,
 		errCh chan<- error,
 		eventsCh *EventChannel,
 	) {
 		defer wg.Done()
+
+		svc, ok := resource.(*cluster.KnativeService)
+		if !ok {
```

Comment:
Thanks for adding this check. Curious - what was the motivation for adding it?

Reply:
deployFunc is refactored to accept any (interface) instead of structural type such as *cluster.KnativeService so the deployFunc can be reused to deploy other resources such as PDB. This line is added to safely cast the resource interface to the concrete structural type.